app/assets/javascripts/covid19/src/domain/entities/Covid19Info.ts

```diff
@@ -166,36 +195,17 @@ export function filterEntities(entities: Entity[], filterState: Covid19Filter):
     );
 }
 
-export function buildPdbRedoValidation(pdbId: Id): PdbRedoValidation {
-    const pdbRedoUrl = `https://pdb-redo.eu/db/${pdbId.toLowerCase()}`;
-
-    return {
-        type: "pdbRedo",
-        externalLink: pdbRedoUrl,
-        queryLink: `/pdb_redo/${pdbId.toLowerCase()}`,
-        badgeColor: "w3-turq",
-    };
-}
-
-export function addPdbValidationToStructure(
-    structure: Structure,
-    validation: PdbValidation
-): Structure {
-    const existingValidations = structure.validations.pdb;
-    const structureContainsValidation = _(existingValidations).some(existingValidation =>
-        _.isEqual(existingValidation, validation)
-    );
-
-    if (!structureContainsValidation) {
-        const pdbValidations = _.concat(existingValidations, [validation]);
-
-        return {
-            ...structure,
-            validations: { ...structure.validations, pdb: pdbValidations },
-        };
-    } else {
-        return structure;
-    }
+export function filterPdbValidations(
+    pdbValidations: PdbValidation[],
+    filterState: Covid19Filter
+): boolean {
+    return filterState.pdbRedo && filterState.isolde && filterState.refmac
+        ? _.some(pdbValidations, ["type", "pdbRedo"]) &&
+              _.some(pdbValidations, ["type", "isolde"]) &&
+              _.some(pdbValidations, ["type", "refmac"])
+        : (filterState.pdbRedo && _.some(pdbValidations, ["type", "pdbRedo"])) ||
+              (filterState.refmac && _.some(pdbValidations, ["type", "refmac"])) ||
+              (filterState.isolde && _.some(pdbValidations, ["type", "isolde"]));
```

Comment:
This code needs some checking. Some notes:

- Make sure the resulting logic operator is AND (as it works for bodies)
- Using lodash getters with strings (`_.some(pdbValidations, ["type", "refmac"])`) is not type safe. Use arrow functions instead.
- Now that I see that the block is an AND. I'd propose the map-approach. Just an idea, unchecked:
```
    const types = pdbValidations.map(val => val.type);
    const filtersEnabled = _.compact([filterState.pdbRedo && "pdbRedo", ...])
    return _.isEmpty(_.difference(..., ...))
```